src/controllers/main/main.ts

```diff
@@ -1103,6 +1103,15 @@ export class MainController extends EventEmitter {
         )
       }
 
+      if (
+        msgAddress === this.selectedAccount.account.addr &&
+        (typedData.primaryType === 'AmbireOperation' || !!typedData.types.AmbireOperation)
+      ) {
+        throw ethErrors.rpc.methodNotSupported(
+          'Forbidden typedData request - signing AmbireOperation is not allowed to prevent fund-draining attacks.'
```

Comment:
change the message to something more generic like "Signing an AmbireOperation is not allowed"